pangea-sdk/service/audit/api.go

```diff
@@ -285,15 +291,36 @@ type LogInput struct {
 	PrevRoot *string `json:"prev_root,omitempty"`
 }
 
-func (i *LogInput) SignEvent(s signer.Signer) error {
-	b := pu.CanonicalizeStruct(&i.Event)
+func (i *LogInput) SignEvent(s signer.Signer, pki map[string]string) error {
+	b, err := pu.CanonicalizeStruct(&i.Event)
+	if err != nil {
+		return err
+	}
 
 	signature, err := s.Sign(b)
 	if err != nil {
 		return err
 	}
+
+	pk, err := s.PublicKey()
+	if err != nil {
+		return err
+	}
+
+	if pki != nil {
+		pki["key"] = pk
```

Comment:
warning: you're not copying the ski struct, so the caller will see the "key" added. I don't know if it may be a problem, but maybe you should copy it just in case

Reply:
You are right, but this struct is just one use struct and internally in Log() function. So user has no access to it.